src/ValidationData.php

```diff
@@ -91,9 +95,13 @@ public function setSubject($subject)
      */
     public function setCurrentTime($currentTime)
     {
-        $this->items['iat'] = (int) $currentTime;
-        $this->items['nbf'] = (int) $currentTime;
-        $this->items['exp'] = (int) $currentTime;
+        foreach ($this->timeFields as $field) {
+            if (isset($this->leeways[$field]) && is_int($this->leeways[$field])) {
```

Comment:
The `is_int()` is to be avoided here - use an assertion where `$this->leeways` is being modified instead, and throw if a non-int is passed in.